### PR TITLE
Migrate dependency management to poetry

### PR DESCRIPTION
**Pull request by legoktm**
Switch dependency management to use poetry, which is much nicer than pip-tools. This is largely based off of
<https://github.com/freedomofpress/securedrop-proxy/pull/122> and applies the same changes to the Makefile and CI.

----
note: this will fail CI until https://github.com/freedomofpress/securedrop-builder/pull/472 is merged.

### CODE REVIEW

File: Makefile
Comment (rocodes) [inline Makefile:14]: this diff makes me _so_ happy. :)

**Comment by legoktm** [comment]
So for test plan, I think just verifying all the various `make test`, etc. commands you run locally during development work and then also doing a package build works. 

I think going through the whole test plan in https://github.com/freedomofpress/securedrop-builder/pull/467 is overkill largely because it was testing all the changes in -builder, which have already been reviewd.

**Comment by rocodes** [comment]
So far so good :)
- [x] visual review
- [x] poetry install succecssful, updated make targets successful 
- [x] building a package from tip of sd-builder works great
I just kicked CI too so we should be seeing lots of :green_heart:

File: .circleci/config.yml
Comment (rocodes) [inline .circleci/config.yml:12]: (Do we want poetry 1.7.0 now that it's available?)

**Review by rocodes — APPROVED** [review]
LGTM. Thanks for making this happen :) (Gonna hold off on merge til end of the day in case you want to sneak the poetry version update in, and we should probably update the dev docs and/or the sdw wiki once some more of these changes land, but :partying_face:)

File: .circleci/config.yml
Comment (legoktm) [reply .circleci/config.yml:12]: Let's stay at 1.6.1 for now since that's what we're using elsewhere and we can do a coordinated bump later. Also it's a bit weird, on bullseye we're using 1.6.1, but on bookworm, the Debian package is at 1.3.2. So we might need to reconcile that too, but we can leave that for a bit later...

**Comment by legoktm** [comment]
(lack of docs issue is https://github.com/freedomofpress/securedrop-dev-docs/issues/103)

**Pull request merged**
